webapp/channels/src/packages/mattermost-redux/src/actions/timezone.ts

```diff
@@ -1,13 +1,17 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import moment from 'moment-timezone';
+
 import {getCurrentTimezoneFull} from 'mattermost-redux/selectors/entities/timezone';
 import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 import type {ActionFuncAsync} from 'mattermost-redux/types/actions';
 
 import {updateMe} from './users';
 
+
 export function autoUpdateTimezone(deviceTimezone: string): ActionFuncAsync {
+    moment.tz.setDefault(deviceTimezone);
```

Comment:
I think having this call outside of the Redux action isn't the best idea. @hmhealey is this okay?

Reply:
I agree with this, however, with my previous change, somehow it is setting the wrong date leading to unexpected functionality discussed above. I just made this change to check if this helps. In any way. I'm somewhat concluded  that setting the default timezone is not the way to fix this, at least here and I've to look for a way to do this via a different method.

Reply:
Not that it matters here any more, but the difference between calling stuff inside the returned function vs outside is minor from what I can tell. Calling it outside makes it happen before Redux middleware has processed the action and calling it inside has it happen while the Redux middleware is doing its thing. Typically, we want to call it inside just for consistency's sake.